pixeltable/dataframe.py

```diff
@@ -287,30 +328,46 @@ def __repr__(self) -> str:
     def _repr_html_(self) -> str:
         return self._description_html()._repr_html_()
 
-    def select(self, *items: exprs.Expr) -> DataFrame:
+    def select(self, *items: exprs.Expr, **named_items : exprs.Expr) -> DataFrame:
         if self.select_list is not None:
             raise exc.Error(f'Select list already specified')
-
-        # analyze select list; wrap literals with the corresponding expressions and update it in place
-        select_list = list(items)
-        for i in range(len(select_list)):
-            expr = items[i]
-            if isinstance(expr, dict):
-                select_list[i] = expr = exprs.InlineDict(expr)
-            elif isinstance(expr, list):
-                select_list[i] = expr = exprs.InlineArray(tuple(expr))
-            elif not isinstance(expr, exprs.Expr):
-                select_list[i] = expr = exprs.Literal(expr)
+        for (alias, _) in named_items.items():
+            if not isinstance(alias, str) or not alias.isidentifier():
+                raise exc.Error(f'Invalid alias: {alias}')
+        base_list = [(expr, None) for expr in items] + [(expr, k) for (k,expr) in named_items.items()]
+        if len(base_list) == 0:
+            raise exc.Error(f'Empty select list')
+        # analyze select list; wrap literals with the corresponding expressions
+        select_list = []
+        for raw_expr, alias in base_list:
+            if isinstance(raw_expr, exprs.Expr):
+                select_list.append((raw_expr, alias))
+            elif isinstance(raw_expr, dict):
+                select_list.append((exprs.InlineDict(raw_expr), alias))
+            elif isinstance(raw_expr, list):
+                select_list.append((exprs.InlineArray(raw_expr), alias))
+            else:
+                select_list.append((exprs.Literal(raw_expr), alias))
+            expr = select_list[-1][0]
             if expr.col_type.is_invalid_type():
-                raise exc.Error(f'Invalid type: {expr}')
+                raise exc.Error(f'Invalid type: {raw_expr}')
             # TODO: check that ColumnRefs in expr refer to self.tbl
+        # check user provided aliases would not conflict among themselves 
+        # or with auto-generated ones
+        seen = set() 
+        for (_, alias) in _expand_select_list(self.tbl, select_list):
+            if alias in seen:
+                raise exc.Error(f'Repeated alias: {alias}')
```

Comment:
Might not be obvious to a user what the two offending expressions are. Better to list those explicitly.